src/app/wallet/wallet.component.ts

```diff
@@ -55,6 +59,106 @@ export class WalletComponent implements OnInit {
+      this.sortHodlings(this.usersYouReceived);        
+      this.sortedFromHighToLow = 0;   
+    } else {
+      this.sortHodlingsToHigh(this.usersYouPurchased);
+      this.sortHodlingsToHigh(this.usersYouReceived);           
+      this.sortedFromHighToLow = 1;    
+    }
+    this.sortedPriceFromHighToLow = -1;
+    this.sortedUsernameFromHighToLow = -1;
+  }
+
+  // sort by coin price
+
+  sortHodlingsPriceToHigh(hodlings: BalanceEntryResponse[]): void {
+
+    hodlings.sort((a: BalanceEntryResponse, b: BalanceEntryResponse) => {
+      return (
+        this.globalVars.bitcloutNanosYouWouldGetIfYouSold(a.ProfileEntryResponse.CoinPriceBitCloutNanos, a.ProfileEntryResponse.CoinEntry) -
+        this.globalVars.bitcloutNanosYouWouldGetIfYouSold(b.ProfileEntryResponse.CoinPriceBitCloutNanos, b.ProfileEntryResponse.CoinEntry) 
+      );
+    });
+  }  
+
+  sortHodlingsPriceToLow(hodlings: BalanceEntryResponse[]): void {
+
+    hodlings.sort((a: BalanceEntryResponse, b: BalanceEntryResponse) => {
+      return (
+        this.globalVars.bitcloutNanosYouWouldGetIfYouSold(b.ProfileEntryResponse.CoinPriceBitCloutNanos, b.ProfileEntryResponse.CoinEntry) -
+        this.globalVars.bitcloutNanosYouWouldGetIfYouSold(a.ProfileEntryResponse.CoinPriceBitCloutNanos, a.ProfileEntryResponse.CoinEntry)
+      );
+    });
+  }    
+
+  toggleSortPrice(){
+    if(this.sortedPriceFromHighToLow){
+      this.sortHodlingsPriceToLow(this.usersYouPurchased);
+      this.sortHodlingsPriceToLow(this.usersYouReceived);         
+      this.sortedPriceFromHighToLow = 0;
+    } else {
+      this.sortHodlingsPriceToHigh(this.usersYouPurchased);
+      this.sortHodlingsPriceToHigh(this.usersYouReceived);          
+      this.sortedPriceFromHighToLow = 1;    
+    }
+    this.sortedFromHighToLow = - 1;
+    this.sortedUsernameFromHighToLow = -1;   
+  }
+
+  // sort by username
+  sortHodlingsUsernameToHigh(hodlings: BalanceEntryResponse[]): void {
+
+    hodlings.sort((a: BalanceEntryResponse, b: BalanceEntryResponse) => {
+      return(
+        a.ProfileEntryResponse.Username.localeCompare(b.ProfileEntryResponse.Username)
+      )
+    });
+
+  }  
+
+  sortHodlingsUsernameToLow(hodlings: BalanceEntryResponse[]): void {
+
+    hodlings.sort((a: BalanceEntryResponse, b: BalanceEntryResponse) => {
+      return(
+        b.ProfileEntryResponse.Username.localeCompare(a.ProfileEntryResponse.Username)
+      )
+    });    
+
+  }    
+
+  toggleSortUsername(){
+    if(this.sortedUsernameFromHighToLow){
+      this.sortHodlingsUsernameToHigh(this.usersYouPurchased);
+      this.sortHodlingsUsernameToHigh(this.usersYouReceived);      
+      this.sortedUsernameFromHighToLow = 0;
+    } else {
+      this.sortHodlingsUsernameToLow(this.usersYouPurchased);
+      this.sortHodlingsUsernameToLow(this.usersYouReceived);  
+      this.sortedUsernameFromHighToLow = 1;    
+    }
+    this.sortedFromHighToLow = - 1;
+    this.sortedPriceFromHighToLow = -1;  
+  }
```

Comment:
seems like this code is a bit duplicative. Anyway we can refactor such that there is one function that handles the sorting of the wallet?